opensrp-anc/src/main/java/org/smartregister/anc/library/presenter/ProfileFragmentPresenter.java

```diff
@@ -68,10 +68,23 @@ public Facts getImmediatePreviousContact(Map<String, String> clientDetails, Stri
                 JSONObject jsonObject = new JSONObject(attentionFlags);
                 if (jsonObject.length() > 0) {
                     Iterator<String> keys = jsonObject.keys();
-
                     while (keys.hasNext()) {
                         String key = keys.next();
-                        facts.put(key, jsonObject.get(key));
+                        String attentionFlagValue = jsonObject.getString(key);
+                        if (attentionFlagValue.length() != 0 && attentionFlagValue.charAt(0) == '{') {
+                            JSONObject attentionFlagObject = new JSONObject(attentionFlagValue);
+                            if (attentionFlagObject.has("text") && attentionFlagObject.has("key")) {
+                                String translation_text;
+                                translation_text = !attentionFlagObject.getString("text").isEmpty() ? "{{" + attentionFlagObject.getString("text") + "}}" : "";
```

Comment:
Please use the constant defined in the different constants files on the app. I think Nativeforms already have `"text"` & `"key"` defined.

Reply:
Done